### PR TITLE
Cleanup Render Result

### PR DESCRIPTION
**Pull request by wyattjoh**
This improves some of the typings around the `RenderResult` returned during renders. Previously it had a single large metadata object that was shared across both the pages and app render pipelines. To add more type safety, this splits the types used by each of the render pipelines into their own types while still allowing the default `RenderResult` to reference metadata from either render pipeline.

This also improved the flight data generation for app renders. Previously, the promise was inlined, and errors were swallowed. With the advent of improvements in #58779 the postpone errors are no longer returned by the flight generation and are instead handled correctly inside the render by React so it can emit properly postponed flight data.

Besides that there was some whitespace changes, so hiding whitespace differences during review should make it much clearer to review!

### CODE REVIEW

**Comment by wyattjoh** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #58782** <a href="https://app.graphite.dev/github/pr/vercel/next.js/58782?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #58779** <a href="https://app.graphite.dev/github/pr/vercel/next.js/58779?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #58708** <a href="https://app.graphite.dev/github/pr/vercel/next.js/58708?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/58782?utm_source=stack-comment).

**Comment by ijjk** [comment]
## Failing test suites <!--18421461884-->
Commit: 8d1e7a780b507a4b2e75bf1a91d40aed58046c91

`pnpm test-start test/e2e/app-dir/ppr-errors/ppr-errors.test.ts` 
- ppr build errors > outside of a suspense boundary > should fail the build for uncaught errors
- ppr build errors > within a suspense boundary > should fail the build for uncaught prerender errors

<details>
<summary>Expand output</summary>

  ● ppr build errors › within a suspense boundary › should fail the build for uncaught prerender errors

    expect(received).toContain(expected) // indexOf

    Expected substring: "Error occurred prerendering page \"/regular-error-suspense-boundary\"."
    Received string:    "
     ⚠ The Next.js plugin was not detected in your ESLint configuration. See https://nextjs.org/docs/basic-features/eslint#migrating-existing-config·
     ⨯ Prerendering /no-suspense-boundary needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error·
     ⨯ Prerendering / needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error·
     ⨯ Prerendering /re-throwing-error needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error
     ⚠ The following error was thrown during build, and may help identify the source of the issue:
     ⨯ Error: The original error was caught and rethrown.

    > 1 | "use strict";(()=>{var e={};e.id=32,e.ids=[32],e.modules={2934:e=>{e.exports=require("next/dist/client/components/action-async-storage.external.js")},5403:e=>{e.exports=require("next/dist/client/components/request-async-storage.external")},4580:e=>{e.exports=require("next/dist/client/components/request-async-storage.external.js")},4749:e=>{e.exports=require("next/dist/client/components/static-generation-async-storage.external")},5869:e=>{e.exports=require("next/dist/client/components/static-generation-async-storage.external.js")},1790:e=>{e.exports=require("next/dist/compiled/next-server/app-page-experimental.runtime.prod.js")},1447:(e,r,t)=>{t.r(r),t.d(r,{GlobalError:()=>a.a,__next_app__:()=>c,originalPathname:()=>u,pages:()=>d,routeModule:()=>x,tree:()=>l});var n=t(9388),o=t(4445),s=t(9863),a=t.n(s),i=t(6453),p={};for(let e in i)0>["default","tree","pages","GlobalError","originalPathname","__next_app__","routeModule"].indexOf(e)&&(p[e]=()=>i[e]);t.d(r,p);let l=["",{children:["re-throwing-error",{children:["__PAGE__",{},{page:[()=>Promise.resolve().then(t.bind(t,8083)),"/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/re-throwing-error/page.jsx"]}]},{}]},{layout:[()=>Promise.resolve().then(t.bind(t,9698)),"/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/layout.jsx"],"not-found":[()=>Promise.resolve().then(t.t.bind(t,5564,23)),"next/dist/client/components/not-found-error"]}],d=["/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/re-throwing-error/page.jsx"],u="/re-throwing-error/page",c={require:t,loadChunk:()=>Promise.resolve()},x=new n.AppPageRouteModule({definition:{kind:o.x.APP_PAGE,page:"/re-throwing-error/page",pathname:"/re-throwing-error",bundlePath:"",filename:"",appPaths:[]},userland:{loaderTree:l}})},8083:(e,r,t)=>{t.r(r),t.d(r,{default:()=>a});var n=t(1529),o=t(5203),s=t(4843);async function a(){return n.jsx(o.Suspense,{fallback:n.jsx("div",{children:"Loading..."}),children:n.jsx(i,{})})}async function i(){try{(0,s.cookies)()}catch(e){throw Error("The original error was caught and rethrown.")}return null}}};var r=require("../../webpack-runtime.js");r.C(e);var t=e=>r(r.s=e),n=r.X(0,[333,396,843],()=>t(1447));module.exports=n})();
        |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ^

      at i (e2e/app-dir/ppr-errors/.next/server/app/re-throwing-error/page.js:1:2065)
      at eP (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:159191)
      at ../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:171056
      at Object.toJSON (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176001)
          at stringify (<anonymous>)
      at eH (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:162810)
      at eV (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:163355)
      at Timeout._onTimeout (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:160166)
      at process.processTimers (../node:internal/timers:514:7)·
       ⨯ Prerendering /no-suspense-boundary-re-throwing-error needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error
       ⚠ The following error was thrown during build, and may help identify the source of the issue:
       ⨯ Error: Throwing a new error from 'no-suspense-boundary-re-throwing-error'
      at s (e2e/app-dir/ppr-errors/.next/server/app/no-suspense-boundary-re-throwing-error/page.js:1:2082)
      at eP (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:159191)
      at ../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:171056
      at Array.toJSON (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176001)
          at stringify (<anonymous>)
      at eH (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:162810)
      at eV (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:163355)
      at Timeout._onTimeout (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176761)
      "
      at Object.toContain (e2e/app-dir/ppr-errors/ppr-errors.test.ts:18:22)

  ● ppr build errors › outside of a suspense boundary › should fail the build for uncaught errors

    expect(received).toContain(expected) // indexOf

    Expected substring: "Error occurred prerendering page \"/regular-error\"."
    Received string:    "
     ⚠ The Next.js plugin was not detected in your ESLint configuration. See https://nextjs.org/docs/basic-features/eslint#migrating-existing-config·
     ⨯ Prerendering /no-suspense-boundary needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error·
     ⨯ Prerendering / needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error·
     ⨯ Prerendering /re-throwing-error needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error
     ⚠ The following error was thrown during build, and may help identify the source of the issue:
     ⨯ Error: The original error was caught and rethrown.

    > 1 | "use strict";(()=>{var e={};e.id=32,e.ids=[32],e.modules={2934:e=>{e.exports=require("next/dist/client/components/action-async-storage.external.js")},5403:e=>{e.exports=require("next/dist/client/components/request-async-storage.external")},4580:e=>{e.exports=require("next/dist/client/components/request-async-storage.external.js")},4749:e=>{e.exports=require("next/dist/client/components/static-generation-async-storage.external")},5869:e=>{e.exports=require("next/dist/client/components/static-generation-async-storage.external.js")},1790:e=>{e.exports=require("next/dist/compiled/next-server/app-page-experimental.runtime.prod.js")},1447:(e,r,t)=>{t.r(r),t.d(r,{GlobalError:()=>a.a,__next_app__:()=>c,originalPathname:()=>u,pages:()=>d,routeModule:()=>x,tree:()=>l});var n=t(9388),o=t(4445),s=t(9863),a=t.n(s),i=t(6453),p={};for(let e in i)0>["default","tree","pages","GlobalError","originalPathname","__next_app__","routeModule"].indexOf(e)&&(p[e]=()=>i[e]);t.d(r,p);let l=["",{children:["re-throwing-error",{children:["__PAGE__",{},{page:[()=>Promise.resolve().then(t.bind(t,8083)),"/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/re-throwing-error/page.jsx"]}]},{}]},{layout:[()=>Promise.resolve().then(t.bind(t,9698)),"/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/layout.jsx"],"not-found":[()=>Promise.resolve().then(t.t.bind(t,5564,23)),"next/dist/client/components/not-found-error"]}],d=["/root/actions-runner/_work/next.js/next.js/test/e2e/app-dir/ppr-errors/app/re-throwing-error/page.jsx"],u="/re-throwing-error/page",c={require:t,loadChunk:()=>Promise.resolve()},x=new n.AppPageRouteModule({definition:{kind:o.x.APP_PAGE,page:"/re-throwing-error/page",pathname:"/re-throwing-error",bundlePath:"",filename:"",appPaths:[]},userland:{loaderTree:l}})},8083:(e,r,t)=>{t.r(r),t.d(r,{default:()=>a});var n=t(1529),o=t(5203),s=t(4843);async function a(){return n.jsx(o.Suspense,{fallback:n.jsx("div",{children:"Loading..."}),children:n.jsx(i,{})})}async function i(){try{(0,s.cookies)()}catch(e){throw Error("The original error was caught and rethrown.")}return null}}};var r=require("../../webpack-runtime.js");r.C(e);var t=e=>r(r.s=e),n=r.X(0,[333,396,843],()=>t(1447));module.exports=n})();
        |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ^

      at i (e2e/app-dir/ppr-errors/.next/server/app/re-throwing-error/page.js:1:2065)
      at eP (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:159191)
      at ../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:171056
      at Object.toJSON (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176001)
          at stringify (<anonymous>)
      at eH (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:162810)
      at eV (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:163355)
      at Timeout._onTimeout (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:160166)
      at process.processTimers (../node:internal/timers:514:7)·
       ⨯ Prerendering /no-suspense-boundary-re-throwing-error needs to partially bail out because something dynamic was used. React throws a special object to indicate where we need to bail out but it was caught by a try/catch or a Promise was not awaited. These special objects should not be caught by your own try/catch. Learn more: https://nextjs.org/docs/messages/ppr-caught-error
       ⚠ The following error was thrown during build, and may help identify the source of the issue:
       ⨯ Error: Throwing a new error from 'no-suspense-boundary-re-throwing-error'
      at s (e2e/app-dir/ppr-errors/.next/server/app/no-suspense-boundary-re-throwing-error/page.js:1:2082)
      at eP (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:159191)
      at ../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:171056
      at Array.toJSON (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176001)
          at stringify (<anonymous>)
      at eH (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:162810)
      at eV (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:163355)
      at Timeout._onTimeout (../packages/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:12:176761)
      "
      at Object.toContain (e2e/app-dir/ppr-errors/ppr-errors.test.ts:52:22)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| buildDuration | 11.2s | 11.1s | N/A |
| buildDurationCached | 6.2s | 6.2s | N/A |
| nodeModulesSize | 199 MB | 199 MB | ⚠️ +21.4 kB |
| nextStartRea..uration (ms) | 410ms | 406ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| 199-HASH.js gzip | 28.7 kB | 28.7 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 494.HASH.js gzip | 180 B | 181 B | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 239 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 100 kB | 100 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 501 B | 503 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | ✓ |
| edge-ssr-HASH.js gzip | 253 B | 255 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.28 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 311 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.17 kB | 3.17 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 483 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| index.html gzip | 527 B | 526 B | N/A |
| link.html gzip | 539 B | 540 B | N/A |
| withRouter.html gzip | 524 B | 521 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| edge-ssr.js gzip | 92.4 kB | 92.4 kB | N/A |
| page.js gzip | 145 kB | 145 kB | ⚠️ +123 B |
| Overall change | 145 kB | 145 kB | ⚠️ +123 B |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 624 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 24.8 kB | 24.8 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js chore/render-result-cleanup | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 167 kB | 167 kB | N/A |
| app-page-exp..prod.js gzip | 93.3 kB | 93.4 kB | N/A |
| app-page-tur..prod.js gzip | 94 kB | 94.1 kB | N/A |
| app-page-tur..prod.js gzip | 88.6 kB | 88.7 kB | N/A |
| app-page.run...dev.js gzip | 137 kB | 137 kB | N/A |
| app-page.run..prod.js gzip | 87.9 kB | 88 kB | ⚠️ +103 B |
| app-route-ex...dev.js gzip | 23.8 kB | 23.8 kB | N/A |
| app-route-ex..prod.js gzip | 16.5 kB | 16.4 kB | N/A |
| app-route-tu..prod.js gzip | 16.5 kB | 16.4 kB | N/A |
| app-route-tu..prod.js gzip | 16 kB | 16 kB | N/A |
| app-route.ru...dev.js gzip | 23.2 kB | 23.2 kB | N/A |
| app-route.ru..prod.js gzip | 16 kB | 16 kB | N/A |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.8 kB | 21.9 kB | N/A |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | N/A |
| pages.runtim..prod.js gzip | 21.8 kB | 21.9 kB | N/A |
| server.runti..prod.js gzip | 49.1 kB | 49.1 kB | N/A |
| Overall change | 116 kB | 116 kB | ⚠️ +103 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>199-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 8cac08ea7fc1b3b64a02a95c64529f3c7bde9c6b

**Review by ztanner — DISMISSED** [review]

File: packages/next/src/export/routes/app-page.ts [hunk 1]
```diff
@@ -128,10 +129,8 @@ export async function exportAppPage(
 
     const html = result.toUnchunkedString()
 
-    const {
-      metadata: { pageData, revalidate = false, postponed, fetchTags },
-    } = result
     const { metadata } = result
+    const { flightData, revalidate = false, postponed, fetchTags } = metadata
```
Comment (feedthejim) [inline packages/next/src/export/routes/app-page.ts:133]: non blocking nit: there's already a concept of metadata in the app router, could we perhaps chose another more explicit name? `prerenderingMetadata`

**Review by timneutkens — DISMISSED** [review]

**Review by ztanner — APPROVED** [review]

**Review by timneutkens — APPROVED** [review]

**Pull request merged**